seqr/views/utils/test_utils.py

```diff
@@ -91,6 +91,93 @@ def _check_login(self, url, permission_level, request_data=None):
 
         self.login_staff_user()
 
+
+ANVIL_WORKSPACES = [{
+    'workspace_namespace': 'test_namespace',
+    'workspace_name': 'anvil-1kg project n\u00e5me with uni\u00e7\u00f8de',
+    'public': False,
+    'acl': {
+        'test_user_manager@test.com': {
+            "accessLevel": "WRITER",
+            "pending": False,
+            "canShare": True,
+            "canCompute": True
+        },
+        'test_user_no_staff@test.com': {
+            "accessLevel": "READER",
+            "pending": False,
+            "canShare": True,
+            "canCompute": True
+        }
+    }
+}, {
+    'workspace_namespace': 'test_namespace',
+    'workspace_name': 'anvil-empty project',
+    'public': False,
+    'acl': {
+        'test_user_manager@test.com': {
+            "accessLevel": "WRITER",
+            "pending": False,
+            "canShare": True,
+            "canCompute": True
+        },
+        'test_user_no_staff@test.com': {
+            "accessLevel": "READER",
+            "pending": False,
+            "canShare": True,
+            "canCompute": True
+        }
+    }
+}, {
+    'workspace_namespace': 'test_namespace',
+    'workspace_name': 'anvil-project 1000 Genomes Demo',
+    'public': False,
+    'acl': {
+        'test_user_manager@test.com': {
+            "accessLevel": "WRITER",
+            "pending": False,
+            "canShare": True,
+            "canCompute": True
+        }
+    }
+}
+]
+
+
+def get_ws_acl_side_effect(url):
+    workspace_namespace, workspace_name = url.split('/')[2:4]
+    wss = filter(lambda x: x['workspace_namespace'] == workspace_namespace and x['workspace_name'] == workspace_name, ANVIL_WORKSPACES)
+    wss = list(wss)
+    return wss[0]['acl'] if wss else {}
+
+
+def get_workspaces_side_effect(user, fields):
+    return [
+        {
+            'public': ws['public'],
+            'accessLevel': ws['acl'][user.email]['accessLevel'],
+            'workspace':{
+                'namespace': ws['workspace_namespace'],
+                'name': ws['workspace_name']
+            }
+        } for ws in ANVIL_WORKSPACES if user.email in ws['acl'].keys()
+    ]
+
```

Comment:
I really like these side effect methods. It makes it super clear how the permissions are set/ decided and it seems really simple to maintain and extend